enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -212,6 +224,168 @@ These changes have already been implemented, and the initial PR for them can be
 
+#### authorization.openshift.io/RestrictSubjectBindings admission plugin considerations
+
+OpenShift's default admission plugin `authorization.openshift.io/RestrictSubjectBindings`
+utilizes the OAuth stack, specifically the `User` and `Group` APIs, to ensure `RoleBinding`
+subjects do not violate any restrictions in the namespace they are going to be created in.
+These restrictions are specified by `RoleBindingRestriction` custom resources.
+
+Removing the OAuth stack entirely may result in the `authorization.openshift.io/RestrictSubjectBindings` admission plugin not functioning as expected.
+
+Anticipated impacts are:
+
+- Logs in the KAS reflecting that informers for OAuth APIs could not be started
+- Errors, that are not actionable from the perspective of users, occurring during validation of `RoleBinding` resources when a `RoleBindingRestriction` resource in the same namespace specifies user/group restrictions. These errors prevent the creation of `RoleBinding`s in a namespace, making it impossible to manage permissions in the namespace.
+
+These impacts may cause concern for cluster administrators that their cluster is in an unhealthy state and may not be easily correlated to taking down the OAuth stack.
+
+To ensure a holistic change in cluster behavior when removing the OAuth stack, and reduce potentially concerning behavior in the control plane components, it is proposed that changes are made to:
+
+- Remove the `rolebindingrestrictions.authorization.openshift.io` CustomResourceDefinition
+- Disable the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins as they are reliant on the `RoleBindingRestriction` resource and/or the OAuth stack being present
+
+##### Changes to the kube-apiserver
+
+The OpenShift-specific patch to the kube-apiserver that adds this admission plugin is found here:
+https://github.com/openshift/kubernetes/blob/3c62f738ce74a624d46b4f73f25d6c15b3a80a2b/openshift-kube-apiserver/openshiftkubeapiserver/patch.go#L71
+
+In order to prevent misleading logs about informers that failed to start or errors related to creating `RoleBinding` resources, the following changes to this patch are to be made:
+
+- `Group` informer creation and configuration is moved into the `authorization.openshift.io/RestrictSubjectBindings` admission plugin initialization process
+- Configuration options are added to the `authorization.openshift.io/RestrictSubjectBindings` admission plugin, allowing for external components to specify the desired state of the OAuth stack.
+
+The configuration option proposed is as follows:
```

Comment:
I'm sure I'm being dense but... why do we need this plugin configuration? KASO writes it but what reads it?  It seems like the KAS configuration is still controlled by {enable|disable}-admission-plugins.

Reply:
The plugin configuration is read by the kube-apiserver during startup and is passed to the admission plugin. We are still explicitly enabling/disabling the admission plugin(s) to be more easily viewable and understandable that we are intentionally disabling the plugins.

Disabling an admission plugin solely with the `--disable-admission-plugins` flag doesn't seem to prevent initialization of the plugin which, in this case, attempts to configure and start informers for an API that will not be present on the cluster. This isn't in and of itself a major issue, but results in a lot of suspicious and noisy logs. Adding the plugin configuration piece allows us to prevent the problematic initialization as the configuration step happens prior to the initialization step.

Reply:
> Disabling an admission plugin solely with the `--disable-admission-plugins` flag doesn't seem to prevent initialization of the plugin

This seems like a bug with `--disable-admission-plugins` no?  If you disable it but some part of it still runs?  But somehow the plugin configuration is fed in early enough to stop the whole plugin?

Said another way, if `--disable-admission-plugins` actually disabled the plugin (informers included), then we wouldn't need the plugin configuration?

Reply:
I did some additional exploration on this and I was unable to reproduce this behavior which tells me this was a mistake I made during my initial testing of this. I've updated this PR to no longer include the addition of a configuration option for the admission plugin and to instead rely solely on the `--enable-admission-plugins` and `--disable-admission-plugins` flags.